cocotb/regression.py

```diff
@@ -201,13 +204,20 @@ def _discover_tests() -> Iterable[Test]:
 
                     yield test
 
-                # only look in first module for all functions and don't complain if all functions are not found
-                break
+                # Clear not_found_tests for the next module search
+                tests = not_found_tests.copy()
+                not_found_tests.clear()
 
-            # auto-discover
-            for thing in vars(module).values():
-                if isinstance(thing, Test):
-                    yield thing
+            else:
+                # auto-discover
+                for thing in vars(module).values():
+                    if isinstance(thing, Test):
+                        yield thing
+
+        # If any test were not found in any module, raise an error
+        if tests:
+            _logger.error("Requested test(s) %s wasn't found in module(s) %s", tests, modules)
+            raise AttributeError("Test(s) %s doesn't exist in %s" % (tests, modules))
```

Comment:
If you have the time could you make another test that covers this line? You can look at [test_failure's Makefile](https://github.com/cocotb/cocotb/blob/master/tests/test_cases/test_failure/Makefile) for idea on how to let the test fail, but the regression pass.

If you don't have time that's okay, just let me know.

Reply:
Yeah, I'll do it tomorrow, if that's not a problem.

Reply:
Added in cf648d4874d0f8c846a7bf27d549ae06f6b3fb89.